snapshot/restore/local_fs_output.go

```diff
@@ -63,8 +86,8 @@ type FilesystemOutput struct {
 	// SkipTimes when set to true causes restore to skip restoring modification times.
 	SkipTimes bool `json:"skipTimes"`
 
-	// Sparse when set to true causes restore files sparsely-not writing any holes (zero regions) to disk.
-	Sparse bool `json:"sparse"`
+	// Copier is the StreamCopier to use for copying the actual bit stream to output.
+	Copier StreamCopier `json:"copier"`
```

Comment:
this won't work, unfortunately. when serialized to JSON.

Since FilesystemOutput is passed verbatim from the UI, we need to pass `sparse` as boolean and resolve it later.

Reply:
Ah good point, I haven't thought of implementing this in UI, and should. Will do

Reply:
Reintroducing `Sparse` as an exported field, and made `Copier` private, to be assigned on the first call to `o.WriteFile()`.

Reply:
Correction, it looks like initializing copier upon the first call to `Writefile()` is not threadsafe, and has caused the race detector test to fail. Introducing `Init()` to replace `NewFilesystemOutput()` to do this assignment. Will call `Init()` for both server and CLI paths.